Update manual steps for Macie: create a new classification job (#656)
gruntwork-io/docs: _docs-sources/guides/build-it-yourself/achieve-compliance/deployment-walkthrough/manual-steps.md

:::note

If you are using Steampipe for checking your Compliance status, you should create the job by "selecting specific buckets",
and **not** by "specifying bucket criteria". Steampipe fetches the bucket list that's being analyzed by Macie, so if you
specify a filter for finding the buckets, Steampipe will not match which bucket are being analyzed. [See Steampipe's
query for finding the buckets analyzed by Macie](https://github.com/turbot/steampipe-mod-aws-compliance/blob/c7cea47662c03f4cc4a84a17e41872e8ace611dc/query/s3/s3_bucket_protected_by_macie.sql#L6-L7).

:::

You can use either the [AWS Console](https://docs.aws.amazon.com/macie/latest/user/discovery-jobs-create.html), or the
[AWS CLI](https://docs.aws.amazon.com/de_de/cli/latest/reference/macie2/create-classification-job.html#create-classification-job) to make the change.
The script below will fetch a list of all buckets in an AWS account and create a new classification job for them to be weekly

> analyzed. The script needs to be executed while authenticated in AWS.

Reviewer: NIT: specify the script depends on `aws` and `jq`. Specify you need to do this in _each_ AWS account.

Author: Fixed!

Follow the steps in the [AWS
docs](https://docs.aws.amazon.com/sns/latest/dg/sns-http-https-endpoint-as-subscriber.html) on how to add an HTTPS endpoint as a subscriber to the alerts.

## 6. Add existing buckets to a new Macie classification job.

Macie is a new AWS service that uses machine learning (ML) and pattern matching to discover and help protect your sensitive

> data. The Landing Zone solution already configured Macie in all deployed accounts. The last step is to setup S3 buckets to be analyzed.

Reviewer: ```suggestion
data. The Landing Zone solution already configured Macie in all deployed accounts. The last step is to specify the S3 buckets to be analyzed.
```

Reviewer: What does the CIS 1.5 benchmark recommend? Do you configure _all_ S3 buckets to be analyzed? Or just some? What does the Steampipe benchmark verify?

If it's all buckets, we need our instructions to say this clearly and explicitly, and explaine how to do this in each AWS account and each AWS region.

Author: >What does the CIS 1.5 benchmark recommend? Do you configure all S3 buckets to be analyzed? Or just some? >What does the Steampipe benchmark verify?

The recommendation says:
>  >Select the check box for each bucket that you want Macie to analyze as part of the job

So not all buckets need to be analyzed. The Steampipe benchmark will fail if not _all_ buckets are being analyzed.

>If it's all buckets, we need our instructions to say this clearly and explicitly, and explaine how to do this in each AWS account and each AWS region.

Gotcha. Macie is a global resource, so at least this is once per account.

I will update the docs!

To set up Macie to analyze the desired S3 buckets, you’ll need to create a **Macie classification job**. There is a bug
in the [terraform-provider-aws](https://github.com/hashicorp/terraform-provider-aws/issues/20726), where the `aws_macie2_classification_job`

> can't be updated. Therefore, we ask you to manually create a new Classification Job and add all desired buckets.

Reviewer: ```suggestion
can't be updated. Therefore, until that bug is fixed (it has been open more than 2 years), we ask you to manually create a new Classification Job and add all desired buckets.
```

:::note

If you are using Steampipe for checking your Compliance status, you should create the job by selecting a list of buckets,

> and **not** by bucket criterea. Steampipe fetchs the bucket list that's being analyzed by Macie, so if you specify critereas

Reviewer: ```suggestion
and **not** by bucket criterea. Steampipe fetches the bucket list that's being analyzed by Macie, so if you specify criteria
```

> for the job, Steampipe will not match that the bucket is indeed being analyzed.

Reviewer: ```suggestion
for the job, Steampipe will not match that the bucket that is being analyzed.
```

Reviewer: TBH, I'm not sure what this note means. Can you give an example of a `steampipe` CLI command that does this?

Author: I re-phrased it:
>If you are using Steampipe for checking your Compliance status, you should create the job by "selecting specific buckets", and **not** by "specifying bucket criteria". Steampipe fetches the bucket list that's being analyzed by Macie, so if you specify a filter for finding the buckets, Steampipe will not match which bucket are being analyzed. [See Steampipe's query for finding the buckets analyzed by Macie](https://github.com/turbot/steampipe-mod-aws-compliance/blob/c7cea47662c03f4cc4a84a17e41872e8ace611dc/query/s3/s3_bucket_protected_by_macie.sql#L6-L7).

The issue is that there are two forms of Macie analyzing buckets: if they are in a list (which our terraform module implements today), or if they are matched in a filter (which we are tracking in https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/issues/204). Today I learned that the Steampipe query will only see a bucket as being analyzed by macie, if they are explicit in a list. So the good UX for the users is actually the way that Steampipe won't pass the compliance check.

Reviewer: Roger, thanks for the explanation!

:::

You can use either the [AWS Console](https://docs.aws.amazon.com/macie/latest/user/discovery-jobs-create.html), or the

> [AWS CLI](https://docs.aws.amazon.com/de_de/cli/latest/reference/macie2/create-classification-job.html#create-classification-job) to make the change.

Reviewer: Does the AWS CLI do the proper "updates"? That is, it doesn't have the same bug that the TF AWS provider does?

Author: In this case, a new classification job is being created, instead of updating the existing one.
